src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4282,10 +4282,10 @@ void os::init(void) {
 
   clock_tics_per_sec = sysconf(_SC_CLK_TCK);
 
-  int page_size = sysconf(_SC_PAGESIZE);
+  size_t page_size = (size_t) sysconf(_SC_PAGESIZE);
```

Comment:
This cast voids the check for negative return values below. Maybe check this value first, then cast it to a size_t?